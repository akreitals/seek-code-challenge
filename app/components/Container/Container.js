import styled from 'styled-components';
import PropTypes from 'prop-types';
import theme from '../theme';

const maxWidth = props =>
    props.maxWidth
        ? { maxWidth: `${props.maxWidth}px` }
        : { maxWidth: props.theme.maxContainerWidth };

const Container = styled.div`
    margin-left: auto;
    margin-right: auto;
    padding: 0 ${theme.space[2]}px;

    ${maxWidth};
`;

Container.displayName = 'Container';

Container.propTypes = {
    maxWidth: PropTypes.number
};

Container.defaultProps = {
    theme
};

export default Container;
