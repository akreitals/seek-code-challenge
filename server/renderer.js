import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
