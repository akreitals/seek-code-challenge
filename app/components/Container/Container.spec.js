import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Container from './Container';

describe('Container', () => {
    test('should render a <Container /> component with default theme max-width', () => {
        const json = renderer.create(<Container />).toJSON();
        expect(json).toMatchSnapshot();
    });

    test('maxWidth prop successfully set maxWidth on container', () => {
        const json = renderer.create(<Container maxWidth={1024} />).toJSON();
        expect(json).toMatchSnapshot();
    });
});
