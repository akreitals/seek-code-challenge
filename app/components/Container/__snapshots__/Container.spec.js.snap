// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container maxWidth prop successfully set maxWidth on container 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  padding: 0 8px;
  max-width: 1024px;
}

<div
  className="c0"
/>
`;

exports[`Container should render a <Container /> component with default theme max-width 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  padding: 0 8px;
  max-width: 940px;
}

<div
  className="c0"
/>
`;
